nagios.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+)
+
+// NagiosStatusData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func GetNagiosStatusData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return retrieveNagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "--", "cat", "/var/log/nagios/status.dat")
+	}, project, "Status", outFor, errOutFor)
+}
+
+// NagiosStatusData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func GetNagiosHistoricalData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return retrieveNagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "--", "tar", "-c", "-C", "/var/log/nagios", "archives", "--to-stdout")
+	}, project, "History", outFor, errOutFor)
+}
+
+// A getNagiosHistoricalDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosHistoricalDataCmdFactory func(project, route string) *exec.Cmd
+
+// A getNagiosStatusDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosStatusDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func retrieveNagiosData(cmdFactory getNagiosStatusDataCmdFactory, project, nagiosResourceType string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		nagiosName := "nagios"
+		stdout, stdoutCloser, err := outFor(project, nagiosName+nagiosResourceType)
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, nagiosName+nagiosResourceType)
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		nagiosPods, err := getProjectPodNamesBySubstr(project, nagiosName)
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPods) == 0 {
```

Comment:
Maybe this should not be an error, at least not in the current form of the code, because we are iterating over all projects and looking for a "nagios" pod in there?